src/Stepic/PrintTextPerRole2.java

```diff
@@ -0,0 +1,36 @@
+package Stepic;
+//this task from the stepic site
+//Write a method that will group the lines by role, number them and return the result in the form of ready-made text
+public class PrintTextPerRole2 {
+    public static void main(String[] args) {
+
+    String roles[] ={"Городничий","Аммос Федорович","Артемий Филиппович","Лука Лукич"};
+    String [] textLines={
+            "Городничий: Я пригласил вас, господа, с тем, чтобы сообщить вам пренеприятное известие: к нам едет ревизор.",
+            "Аммос Федорович: Как ревизор?",
+            "Артемий Филиппович: Как ревизор?",
+            "Городничий: Ревизор из Петербурга, инкогнито. И еще с секретным предписаньем.",
+            "Аммос Федорович: Вот те на!",
+            "Артемий Филиппович: Вот не было заботы, так подай!",
+            "Лука Лукич: Господи боже! еще и с секретным предписаньем!"};
+        //array for variables to search by role
+    String[] forSearching=new String[roles.length];
+   //create a variable to return
+        StringBuilder toReturn =new StringBuilder();
+        for(int i =0;i< roles.length;i++){
+            forSearching[i] = roles[i]+":";
+            toReturn.append(forSearching[i]);
+            toReturn.append('\n');
+            for(int j=0;j< textLines.length;j++){
+                if(textLines[j].startsWith(forSearching[i]+ " ")) {
+                    String toReplace = (j+1)+")";
+                    toReturn.append(toReplace);
+                    toReturn.append(textLines[j].substring(forSearching[i].length()));
+                    toReturn.append('\n');
+            }
```

Comment:
I would like to move it to the separate method

Reply:
already remade